go-controller/pkg/ovn/gress_policy.go

```diff
@@ -100,10 +100,18 @@ func (gp *gressPolicy) ensurePeerAddressSet(factory AddressSetFactory) error {
 	return nil
 }
 
-func (gp *gressPolicy) addPeerPod(pod *v1.Pod) error {
+func (gp *gressPolicy) addPeerPod(pod *v1.Pod, oc *Controller) error {
 	ips, err := util.GetAllPodIPs(pod)
 	if err != nil {
-		return err
+		// If we could not gather IPs from the pod itself, check logical port cache
+		portName := podLogicalPortName(pod)
+		portInfo, _ := oc.logicalPortCache.get(portName)
```

Comment:
I assume that on the cache, the logicalPort would always be unique for each pod, isn't it? Otherwise there might be funny actions lying around.

Reply:
yeah the the port in the cache should be unique to the pod. If/When we support multiple ports for a pod we will need to go back and refactor how we name these port entries.